pkg/controller/priorityqueue/priorityqueue.go

```diff
@@ -290,9 +290,15 @@ func (w *priorityqueue[T]) GetWithPriority() (_ T, priority int, shutdown bool)
 
 	w.notifyItemOrWaiterAdded()
 
-	item := <-w.get
-
-	return item.Key, item.Priority, w.shutdown.Load()
+	select {
```

Comment:
This was way to tricky to debug :)

Once PQ was enabled per default, this test started to fail: https://github.com/kubernetes-sigs/controller-runtime/blob/b3590eadfe1bd5521151a79b586a2bd6f376580b/pkg/controller/controller_integration_test.go#L52

* manager stop failed with "failed waiting for all runnable to end within grace period"
* because not all workers of the leader election runnable group stopped
* and they didn't stop because they were stuck in GetWithPriority()

Just tricky to figure out: 
1. which test even caused this (I just got some apiserver stop timeout from a suite)
2. which runnable group did not stop
3. which runnable did not stop
4. why does it not stop (is reconcile blocking or something else)

I guess our logging could use some improvements ...

Reply:
Moved to https://github.com/kubernetes-sigs/controller-runtime/pull/3337

I'll rebase this PR once 3337 merges